### PR TITLE
Architecture decision record for unified Django architecture

### PR DESCRIPTION
**Pull request by tadhg-ohiggins**
New information
Leads to reconsidering
And new approaches.

-----

Add ADR and system diagram for switch to unified Django web application architecture.

### CODE REVIEW

**Comment by tadhg-ohiggins** [comment]
Addresses #553

File: docs/architecture/decisions/0012-switch-to-unified-django-web-app.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Lastly, while ultimately supporting an API-centric path for the submission of single audits remains a goal, it cannot be considered a priority under the current circumstances. There is no current demand among submitters for this, and the deadline means that supporting the interactive submission path must be the priority.
+
+## Decision
+
+We will move the current frontend into Django, use Django to manage state, add whatever rich interaction we need to via as-yet-undetermined JavaScript frameworks, and continue to use USWDS styling and components. This also means we will move to a single repository, something that was already looking likely regardless of this decision.
+
+## Considerations
+
+*   We need to prioritize development speed, and therefore simplicity of development environment. When the frontend required less state management, the approach of having two separate environments, both relatively simple, was feasible. But once one or both of the hit a certain point, the difficulty of managing synchronization between the environments became an impediment.
```
Comment (timoballard) [inline docs/architecture/decisions/0012-switch-to-unified-django-web-app.md:45]: typo: "But once one or both of **them**

File: docs/architecture/decisions/0012-switch-to-unified-django-web-app.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+*   Making the change sooner rather than later means less work to potentially re-do.
+*   Making the change sooner rather than later means a better chance of having the new architecture settled enough for ease of vendor onboarding, which may be happening soon.
+
+## Consequences
+
+*   Better long-term outlook.
+*   Redoing a lot of the frontend work to use Django templates and views.
+*   Django learning for a lot of the team.
+*   Feature work delayed as we make the move.
```
Comment (timoballard) [inline docs/architecture/decisions/0012-switch-to-unified-django-web-app.md:54]: Additional consequence: "Revisiting authentication flow / Login.gov integration (will move from `PKCE` flow to `private_key_jwt`)"

**Review by timoballard — CHANGES_REQUESTED** [review]
Added an additional consequence

File: docs/architecture/decisions/0012-switch-to-unified-django-web-app.md (hunk 1 above)
Comment (tadhg-ohiggins) [reply docs/architecture/decisions/0012-switch-to-unified-django-web-app.md]: Corrected, thanks!

**Review by timoballard — APPROVED** [review]
LGTM

**Review by heymatthenry — APPROVED** [review]
LGTM. I had forgotten about the sessions piece. Yet another reason.

**Pull request merged**